Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,297 @@
+        } else {
+            for (NSUInteger i = start; i < end; i++) {
+                [self _insertObject:[array objectAtIndex:i]];
+            }
+        }
+    }
+    return self;
+}
+
+- (void)_insertObject:(id)object {
+    if ([self containsObject:object]) {
+        return;
+    }
+
+    [_arrayContainer addObject:object];
+    [_setContainer addObject:object];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (NSUInteger)count {
+    return [_arrayContainer count];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSArray*)array {
+    return [[[_NSProxyOrderedSetArray alloc] initWithOrderedSet:self] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSSet*)set {
+    return [[_setContainer copy] autorelease];
+}
+
+/**
+ @Status Interoperable
 */
 - (instancetype)initWithObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (object == nil) {
+        [NSException raise:NSInvalidArgumentException format:@"object can not be nil."];
+    }
+    id ret[] = { object };
+    return [self initWithObjects:ret count:1];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id)firstObj {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithObjects:(id)firstObj, ... {
+    va_list argList;
+    va_start(argList, firstObj);
+    std::vector<id> flatArgs = ConvertVAListToVector((id)firstObj, argList);
+    va_end(argList);
+    return [self initWithObjects:flatArgs.data() count:flatArgs.size()];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)cnt {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    if (objects == nil) {
+        [NSException raise:NSInvalidArgumentException format:@"objects can not be nil."];
+    }
+
+    if (self = [self init]) {
+        for (NSUInteger i = 0; i < count; ++i) {
+            [self _insertObject:objects[i]];
+        }
+    }
+    return self;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (orderedSet == nil) {
+        return [self init];
+    }
+
+    return [self initWithArray:orderedSet->_arrayContainer range:range copyItems:flag];
```

Comment:
Since you're accessing the ivars here, you could just set set/arrayContainer to mutable copies of the other ones :)